src/ThriftSQL/ResultRow.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace ThriftSQL;
+
+class ResultRow implements \ArrayAccess
+{
+  private $_row;
+  private $_schema;
+
+  public function __construct($schema, $row)
+  {
+    $this->_row = $row;
+    $this->_schema = $schema;
+  }
+
+  public function schemaArray()
+  {
+    $r = [];
+    foreach ($this->_schema as $k=>$i) {
+      $r[$k] = $this->_row[$i];
+    }
+    return $r;
+  }
+
+  private function offsetToIndex($offset)
+  {
+    if(is_int($offset)) {
+      return 0 <= $offset && $offset < count($this->_row);
+    } else if (is_string($offset)) {
+      return ($this->_schema[strtolower($offset)] ?? null);
+    }
+    return null;
+  }
+  /**
+   * @inheritDoc
+   */
+  public function offsetExists($offset)
+  {
+    return array_key_exists($this->offsetToIndex($offset), $this->_row);
```

Comment:
Maybe we can just check the schema to see if the key given exists?

Reply:
I think we can support `$resultRow[0]` and `$resultRow['id']` at the same time.